prep-changes

evanfkivi opened the pull request.

evanfkivi committed:
added viewModel

evanfkivi committed:
wtf

evanfkivi committed:
added the following:

-Info screen viewmodel
-Use of stateflows
-Fixed routes w/ type safe nav
-made albums screen items reusable

lucas-livefront commented on app/src/main/java/com/example/beatlesapp/ui/BeatlesViewModel.kt:
```diff
@@ -1,8 +1,23 @@
+    val state: StateFlow<List<AlbumItem>> = _uiState.asStateFlow()
+}
+
+class InfoScreenViewModel(
```
I feel like you have two options here:
1. Put this in its own file.
2. Rename the file to something more generic that might imply there are more than one VM in here.

lucas-livefront commented on build.gradle.kts:
```diff
@@ -3,4 +3,6 @@ plugins {
     alias(libs.plugins.android.application) apply false
     alias(libs.plugins.kotlin.android) apply false
     alias(libs.plugins.kotlin.compose) apply false
+    kotlin("jvm") version "2.0.0" // Or your current Kotlin version
+    kotlin("plugin.serialization") version "2.0.0" // Match Kotlin version
```
Seems like these comments are just copied and pasted from the docs, it is probably a good idea to delete them. It is just a sign of noobiness 😆 cause ideally you will be able to show this off some day.

evanfkivi committed:
added the following:

-Info screen viewmodel
-Use of stateflows
-Fixed routes w/ type safe nav
-made albums screen items reusable

evanfkivi committed:
added the following:

-Info screen viewmodel
-Use of stateflows
-Fixed routes w/ type safe nav
-made albums screen items reusable
